[justinlengch] iP

justinlengch opened the pull request:
# Rahhbot User Guide

## Introduction
Welcome to RahhBot! RahhBot is your friendly virtual assistant designed to make your life easier. Whether you need help managing your tasks, finding information, or just want to chat, RahhBot is here to assist you. This user guide will walk you through the basic functionalities of RahhBot and how to interact with it effectively.

## Getting Started
To start using RahhBot, simply type your queries or commands into the chat window. RahhBot will respond promptly and provide you with the assistance you need. You can ask RahhBot about tasks, deadlines, events, and more.

## Basic Commands
### 1. Creating a Task:
- To create a new task, use the command: todo **task description**.
- Example: todo Do laundry.

### 2. Creating a Deadline Task:
- To create a task with a deadline, use the command: deadline **task description** /by **date**.
- Example: deadline Submit report /by 2024-02-13.

### 3. Creating an Event Task:
- To create an event task, use the command: event **event description** /from **start date** /to **end date**.
- Example: event Team meeting /from 2024-02-27 /to 2024-02-27.

### 4. Listing Tasks:
- To list all tasks, use the command: list.

### 5. Marking Task as Done:
- To mark a task as done, use the command: mark **task index**.
- Tip: Use the list command to find the index of the task you want to mark as done.
- Example: mark 1.

### 6. Deleting a Task:
- To delete a task, use the command: delete **task index**.
- Example: delete 2.

### 7. Finding Tasks:
- To search for tasks containing a specific keyword, use the command: find **keyword>**.
- Example: find meeting.
- It's case-insensitive, so you can use any combination of uppercase and lowercase letters or even partial keywords!

### 8. Exiting RahhBot:
- To exit RahhBot, simply type: bye.

## Additional Features
### Help Command:
- If you need assistance or want to see a list of available commands, type: help.

## Conclusion
That's it! You are now ready to start using RahhBot to manage your tasks and get things done more efficiently. If you have any questions or encounter any issues, feel free to ask RahhBot for help. Enjoy chatting with RahhBot!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

justinlengch committed:
did level 0 increments

justinlengch committed:
added level 1 Echo

justinlengch committed:
level-2 increments

justinlengch committed:
did level 3 increments

justinlengch committed:
made level 4 increments and added a input command identifier

justinlengch committed:
A-TextUiTesting increment

justinlengch committed:
changed back bot name to Duke

justinlengch committed:
added exceptions handling

justinlengch committed:
ArrayList already used early on. Added delete tasks command.

belligerentbeagle commented:
Hi Justin, you might wna capitalize your "p" so that the bot can detect your PR!

soons1 commented on src/main/java/DeadlineTask.java:
```diff
@@ -0,0 +1,57 @@
+        super.updateDoneIcon();
+    }
+    public static String[] getDescription(String[] details) throws DukeException {
+        int byIndex = -1;
```
Would it be clearer to use a more specific name rather than byIndex?

soons1 commented on src/main/java/CommandScanner.java:
```diff
@@ -0,0 +1,37 @@
+    public static String scanCommand(String[] inputs) throws DukeException {
+
+        if (inputs.length > 0) {
+            String firstWord = inputs[0].toUpperCase();
```
Maybe we can use a name like inputCmd instead of firstWord to make it more specific?

soons1 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,85 @@
+
+        while (runBot) {
+            String userInput = scanner.nextLine().trim();
+            String[] words = userInput.split("\\s+");
```
Would using a more specific name other than words be better?

soons1 commented on src/main/java/DeadlineTask.java:
```diff
@@ -0,0 +1,57 @@
+        return new String[]{description, deadlineDate};
+    }
+    @Override
+    public String getDoneIcon() {
```
Very clear method names! It is easy to understand what each method does just by reading the name.

soons1 reviewed (commented):
LGTM 👍 
Issues identified are really minor

justinlengch committed:
did level-7 save, and cleanup Duke.java

justinlengch committed:
Level 8 date time and cleaned up exceptions

justinlengch committed:
added level-8 increments

justinlengch committed:
Merge branch 'branch-Level-7'

justinlengch committed:
Merge branch 'branch-Level-8'

justinlengch committed:
did more oop stuff

justinlengch committed:
Merge branch 'A-MoreOOP'

justinlengch committed:
put .java files in respective packages

justinlengch committed:
added .java files into their respective packages

justinlengch committed:
fix conflict issues

justinlengch committed:
pushed text-ui-test files

justinlengch committed:
Merge remote-tracking branch 'origin/add-gradle-support'

justinlengch committed:
Added Gradle support

justinlengch committed:
Merge branch 'branch-A-Gradle'

justinlengch committed:
created JUnit tests to test 4 non-trivial methods

justinlengch committed:
Merge branch 'branch-A-JUnit'

justinlengch committed:
edit stuff to make jar work

justinlengch committed:
Merge branch 'branch-A-jar'

justinlengch committed:
added javadoc for all methods

justinlengch committed:
edited code based on SonarLint coding standard

justinlengch committed:
Added Find command

justinlengch committed:
Merge branch 'branch-A-JavaDoc'

justinlengch committed:
merge master to branch-A-CodingStandard

justinlengch committed:
Merge branch 'master' into branch-A-JavaDoc

justinlengch committed:
Added remaining javadoc for all methods

justinlengch committed:
Merge branch 'branch-A-JavaDoc'

justinlengch committed:
fix conflict and merge Find to master

justinlengch committed:
started some code for javafx in master

justinlengch committed:
added javaFX gui and change all sout to Strings

justinlengch committed:
Merge branch 'branch-Level-10'

justinlengch committed:
feat: Add handling for NumberFormatException in input parsing

Description:
Added error handling for NumberFormatException to improve robustness in parsing user input. This change ensures that the program gracefully handles cases where input provided by the user cannot be parsed into numerical values, preventing unexpected crashes and improving user experience.

Changes:
- Implemented code to catch NumberFormatException during input parsing

justinlengch committed:
Added some assertions to Parser.java

justinlengch committed:
Merge pull request #2 from justinlengch/branch-A-Assertions

Added some assertions to Parser.java

justinlengch committed:
Added more error handling prevent edge cases from breaking jar file

justinlengch committed:
Merge pull request #3 from justinlengch/branch-A-Assertions

Added more error handling prevent edge cases from breaking jar file

justinlengch committed:
Already edited code quality based on SonarLint rules

justinlengch committed:
Merge pull request #4 from justinlengch/branch-A-CodeQuality

Already edited code quality based on SonarLint rules

justinlengch committed:
Added user guide in readme and Ui image

justinlengch committed:
Changed some readme stuff

justinlengch committed:
Added more changes to readme

justinlengch committed:
added code snippets in readme

justinlengch committed:
added slight changes to readme

justinlengch committed:
small edits

justinlengch committed:
more edit

justinlengch committed:
more edits

justinlengch committed:
alot more changes

justinlengch committed:
few more edits
